lightning/src/chain/channelmonitor.rs

```diff
@@ -605,13 +605,15 @@ impl Readable for ChannelMonitorUpdateStep {
 pub struct ChannelMonitor<ChanSigner: ChannelKeys> {
 	latest_update_id: u64,
 	commitment_transaction_number_obscure_factor: u64,
+	is_outbound: bool,
 
 	destination_script: Script,
 	broadcasted_holder_revokable_script: Option<(Script, PublicKey, PublicKey)>,
 	counterparty_payment_script: Script,
 	shutdown_script: Script,
 
 	keys: ChanSigner,
+	counterparty_pubkeys: ChannelPublicKeys,
```

Comment:
> I don't follow this change at reviewing [ec16cf2](https://github.com/rust-bitcoin/rust-lightning/commit/ec16cf20c5ce7c7f142bd5ae0199ac413dc8eaf4), is it consumed somewhere ?

I removed this field, my mistake.

> Further, I even think we could drop any counterparty pubkeys by extending ChannelKeys with `counterparty_pubkeys` as they're stored their post channel acceptance. ChannelMonitor already have access to key signer, so it's not an issue calling at counterparty script generation.

@ksedgwic and I were thinking about that, but I'm a bit concerned about making the node dependent on the signer for transaction construction.  I wonder if it may make it harder to perform some system-administration operations when the signer is unavailable for some reason.  But otherwise, that's something that can be done in the followup PR.

Reply:
Just fyi, you have access to counterparty keys in the following contexts:

* To claim a revoked `to_local` output on counterparty tx, L1318 in `src/chain/channelmonitor.rs`
* To claim a revoked HTLC output on counterparty tx, L1331 in `src/chain/channelmonitor.rs`
* To claim a valid HTLC output on counterparty tx, L1470 in `src/chain/channelmonitor.rs`
* To claim a revoked output on counterparty HTLC tx, L1502 in `src/chain/channelmonitor.rs`

All those cases assume your monitor is  feeded with blocks (`block_connected`) but I guess in system-administration you will turn off chain watching to avoid a state update ? That said I agree it's hard to scope yet and orthogonal to this PR.